streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -839,6 +845,14 @@ class ZStream[-R, +E, +A](val process: ZManaged[R, E, Pull[R, E, A]]) extends Se
       } yield pull
     }
 
+  /**
+   * Returns a stream whose failures and successes have been lifted into an
+   * `Either`.The resulting stream cannot fail, because the failures have
+   * been exposed as part of the `Either` success case.
```

Comment:
Cool. Add a note that the resulting stream will emit the failure that occured, then end?

Reply:
Ha, right. Is that the correct behaviour? or instead we keep pulling even after error?

Reply:
Yes, that's what we do right now because the contract of `Pull` (currently!) doesn't allow for pulling after a failure.